src/Npgsql/NpgsqlConnector.cs

```diff
@@ -451,11 +450,12 @@ void WriteStartupMessage(string username)
                     "UTF8"
             };
 
-            Debug.Assert(Settings.Database != null);
-            startupParams["database"] = Settings.Database;
-            if (!string.IsNullOrEmpty(Settings.ApplicationName))
+            startupParams["database"] = Settings.Database!;
```

Comment:
We can make this better by passing the non-nullable database down the stack, or introducing another internal non-nullable property. But we can do this later (it's the same as before).